### PR TITLE
Improve loss handling 

### PR DESCRIPTION
**Pull request by arthurbr11**

### CODE REVIEW

File: sentence_transformers/sparse_encoder/losses/SpladeLoss.py [hunk 1]
```diff
@@ -43,13 +32,17 @@ def __init__(
...
-            query_regularizer: The loss that will be used for the query. If lambda_query is not 0 or None and this isn't specified, it will be a FlopLoss.
-
+            loss: The principal loss function to use can be any of the SparseEncoder losses except CSR related losses and flops loss.
+            lambda_corpus: Weight for the corpus regularization term. This term encourages sparsity in the document embeddings.
+                Will be applied to positive documents and all negatives one if some are provided.
+            lambda_query: Weight for the query regularization term. This term encourages sparsity in the query embeddings.
+                If None, no query regularization will be applied, it's not a problem if you are in an inference-free setup or
+                if you are having all_docs=True. Else you should have a lambda_query > 0.
+            all_docs: If True, all input embeddings are treated as documents and regularized together with lambda_corpus.
+                Espacially useful when training with input likes sentence pairs or more.
```
Comment (tomaarsen) [inline sentence_transformers/sparse_encoder/losses/SpladeLoss.py]:
```suggestion
            all_docs: If True, all input embeddings are treated as documents and regularized together with lambda_corpus.
                Especially useful when training with symmetric texts (e.g. pairs of documents) or more.
```
Maybe something like this?

File: sentence_transformers/sparse_encoder/losses/SpladeLoss.py [hunk 2]
```diff
@@ -43,13 +32,17 @@ def __init__(
...
+            threshold: Optional threshold for the number of non-zero elements in the embeddings to be considered in the FlopsLoss.
+                If specified, only embeddings with more than this number of non-zero elements will be considered.
+                This can help to ignore embeddings that are too sparse and may not contribute meaningfully to the loss.
```
Comment (tomaarsen) [inline sentence_transformers/sparse_encoder/losses/SpladeLoss.py:45]:
```suggestion
                This can help to ignore embeddings that are too sparse and may not contribute meaningfully to the loss.

```

File: sentence_transformers/sparse_encoder/losses/SpladeLoss.py [hunk 3]
```diff
@@ -93,7 +93,7 @@ def compute_labels(batch):
         self.loss = loss
         self.regularizer = FlopsLoss(model, threshold=threshold)
         self.lambda_corpus = lambda_corpus
-        self.lambda_query = lambda_corpus
+        self.lambda_query = lambda_query
```
[the remark this answers is not in the document]
Comment (arthurbr11) [reply sentence_transformers/sparse_encoder/losses/SpladeLoss.py:96]: ahah no problem, just saw it thanks to the IDE

**Comment by arthurbr11** [comment]
So do you think we should only have this instead of the threshold ?
